embedded-graphics/src/image/image_bmp.rs

```diff
@@ -129,15 +130,16 @@ where
                 return None;
             }
 
-            let offset = ((h - 1 - y) * w) + x;
+            let offset = (((h - 1 - y) * w) + x) as usize;
 
-            let bit_value = if self.im.bmp.bpp() == 8 {
-                self.image_data[offset as usize] as u16
+            let data = if self.im.bmp.bpp() == 8 {
+                &self.image_data[offset..]
             } else if self.im.bmp.bpp() == 16 {
-                let offset = offset * 2; // * 2 as two bytes per pixel
-
-                (self.image_data[offset as usize] as u16)
-                    | ((self.image_data[(offset + 1) as usize] as u16) << 8)
+                &self.image_data[offset * 2..] // * 2 as two bytes per pixel
+            } else if self.im.bmp.bpp() == 24 {
+                &self.image_data[offset * 3..] // * 3 as two bytes per pixel
+            } else if self.im.bmp.bpp() == 32 {
+                &self.image_data[offset * 4..] // * 4 as two bytes per pixel
```

Comment:
Nit: Can you make this if/else chain a `match` instead? I think that'd look cleaner.